helm_remote/Tiltfile

```diff
@@ -71,29 +119,36 @@ def helm_remote(chart, repo_url='', repo_name='', release_name='', values=[], se
 
     # ======== Initialize
     # -------- targets
-    pull_target = os.path.join(helm_remote_cache_dir, repo_name)
-    if version != '':
-        pull_target = os.path.join(pull_target, version)
-    else:
-        pull_target = os.path.join(pull_target, 'latest')
-
+    pull_target = os.path.join(helm_remote_cache_dir, repo_name, version)
     chart_target = os.path.join(pull_target, chart)
 
-    # -------- commands
-    pull_command = 'helm pull %s/%s --untar --destination %s' % (repo_name, chart, pull_target)
-    if version != '':
-        pull_command += ' --version %s' % version
-    if username != '':
-        pull_command += ' --username %s' % username
-    if password != '':
-        pull_command += ' --password %s' % password
-
-    # ======== Perform Installation
-    local('rm -rf %s' % chart_target,
-          command_bat='if exist %s ( rd /s /q %s )' % (chart_target, chart_target))
-    local(pull_command)
-
-    install_crds(chart, chart_target)
+    cached_chart_exists = os.path.exists(chart_target)
+
+    pull = True
+
+    if cached_chart_exists == True:
+        # Helm chart structure is concrete, we can trust this YAML file to exist
+        cached_chart_details = read_yaml(os.path.join(chart_target, 'Chart.yaml'))
+
+        # check if our local cached chart matches latest remote
+        remote_chart_details = fetch_chart_details(chart, repo_name, (username, password), version)
+        
+        # pull when version mismatch
+        pull = cached_chart_details['version'] != remote_chart_details['version']
+
+
+    if pull == True:
+        # -------- commands
+        pull_command = 'pull %s/%s --untar --destination %s' % (repo_name, chart, pull_target)
+
+        # ======== Perform Installation
+        if cached_chart_exists == True:
+            local('rm -rf %s' % chart_target,
+                  command_bat='if exist %s ( rd /s /q %s )' % (chart_target, chart_target))
+
+        local(build_helm_command(pull_command, (username, password), version))
+
+        install_crds(chart, chart_target)
```

Comment:
the install_crds command needs to be outside the if block, right?

Reply:
@nicks I left it purposefully in the `pull` scope because the result of `install_crds()` would be idempotent, when the helm version version has not changed.

I'm happy to move it outside this scope though if it's preferred

Reply:
Hmmm...I think you're mistaken. The Helm chart is stored locally. The CRDs are stored in the cluster. If you reset the cluster, the Helm version will be the same, but the CRDs will need to be installed still.

Reply:
That's a really valid point, it's not checking the helm chart metadata, and it's possible for two different Tiltfiles to use the same chart, so it's a possible case.
I've moved it outside this scope now